SpatialConvolution.lua

```diff
@@ -155,8 +155,8 @@ function SpatialConvolution:accGradParameters(input, gradOutput, scale)
 end
 
 function SpatialConvolution:type(type,tensorCache)
-   self.finput = torch.Tensor()
-   self.fgradInput = torch.Tensor()
+   self.finput = self.finput and self.weight.new()
+   self.fgradInput = self.fgradInput and self.weight.new()
```

Comment:
I actually think you should do it for `finput` as well (as it was before), because converting an already `forwarded` CPU model to GPU would create a huge (and unnecessary) `finput` buffer.